src/main/scopes/js/metrics/function-metric.ts

```diff
@@ -71,24 +74,48 @@ export class FunctionMetric extends ScopeMetric {
       this.instrumentedPackage.version
     )
 
-    // convert arguments into a fake object to satisfy the substitute api
-    const argsObj: Record<string, unknown> = this.jsFunction.arguments.reduce(
-      (cur, val, index) => ({ ...cur, [index.toString()]: val }),
-      {}
-    )
+    let functionName = this.jsFunction.name
+    const functionAccessPath = this.jsFunction.accessPath
 
-    const anonymizedArguments = substitute(
-      argsObj,
-      Object.keys(argsObj), // all keys are allowed
-      this.allowedArgumentStringValues
-    )
+    // Handle renamed functions
+    if (this.matchingImport.rename !== undefined) {
+      functionName = functionName.replace(this.matchingImport.rename, this.matchingImport.name)
+      // replace the import name in access path
+      functionAccessPath[0] = this.matchingImport.name
+    }
+
+    const subs = new Substitution()
+    // redact complex values
+    functionAccessPath.forEach((segment) => {
+      if (segment instanceof ComplexValue) {
+        functionName = safeStringify(functionName).replace(
+          safeStringify(segment.complexValue),
+          subs.put(segment)
+        )
+      }
+    })
+
+    // redact "all" import
+    if (this.matchingImport.isAll) {
+      functionAccessPath[0] = subs.put(this.matchingImport.name)
```

Comment:
This unfortunately still mutates the field on the class 😬 
Since `functionAccessPath` is a reference to `this.jsFunction.accessPath`, modifying its elements will modify the original array.